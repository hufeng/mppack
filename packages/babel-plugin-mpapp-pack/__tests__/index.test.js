const babel = require('babel-core');
const pack = require('../index');

const testCode = `
//test mixin
const TabMixin = Mixin({
  data: {
    tabIndex: 0
  },

  onHandleTabIndex(tabIndex) {
   this.setData({
     tabIndex
   });
   this.$spliceData({})
  }
});

//test plain javascript object
const pojo = {
  data: {
    test: 0
  },

  sayHello() {
    this.setData({})
    this.$spliceData({})
  }
};

//test mue object
Mue({
  data: {
    list: []
  },

  sayHello() {
    this.setData({});
    this.$spliceData({})
  }
});

Action({
  sayHello() {
    this.setData({});
    this.$spliceData({})
  }
})
`;

it('test setData and spliceData', () => {
  const { code } = babel.transform(testCode, {
    plugins: [pack]
  });
  expect(code).toMatchSnapshot();
});

it('test async/await', () => {
  const sourceCode = `
  var _onLoad = _asyncToGenerator(
    /*#__PURE__*/
    regeneratorRuntime.mark(function _callee() {
      var _this = this;

      var data;
      return regeneratorRuntime.wrap(function _callee$(_context) {
        while (1) {
          switch (_context.prev = _context.next) {
            case 0:
              app.getUserInfo(function (userInfo) {
                _this.setState({
                  userInfo: userInfo
                });
              });
              console.log('onLoading...');
              _context.next = 4;
              return webapi.fetchData();

            case 4:
              data = _context.sent;
              this.setState({
                todo: data
              });

            case 7:
            case "end":
              return _context.stop();
          }
        }
      }, _callee, this);
    }));
    var _onLoad1 = _asyncToGenerator(
      /*#__PURE__*/
      regeneratorRuntime.mark(function _callee() {
        var _this = this;

        var data;
        return regeneratorRuntime.wrap(function _callee$(_context) {
          while (1) {
            switch (_context.prev = _context.next) {
              case 0:
                app.getUserInfo(function (userInfo) {
                  _this.setState({
                    userInfo: userInfo
                  });
                });
                console.log('onLoading...');
                _context.next = 4;
                return webapi.fetchData();

              case 4:
                data = _context.sent;
                this.setState({
                  todo: data
                });

              case 7:
              case "end":
                return _context.stop();
            }
          }
        }, _callee, this);
      }));
  `;

  const { code } = babel.transform(sourceCode, {
    plugins: [pack]
  });

  expect(code).toMatchSnapshot();
});
