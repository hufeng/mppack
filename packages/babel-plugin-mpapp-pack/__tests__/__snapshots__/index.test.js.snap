// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test async/await 1`] = `
"import _regeneratorRuntime from 'regenerator-runtime';

var _onLoad = _asyncToGenerator(
/*#__PURE__*/
_regeneratorRuntime.mark(function _callee() {
  var _this = this;

  var data;
  return _regeneratorRuntime.wrap(function _callee$(_context) {
    while (1) {
      switch (_context.prev = _context.next) {
        case 0:
          app.getUserInfo(function (userInfo) {
            _this.setState({
              userInfo: userInfo
            });
          });
          console.log('onLoading...');
          _context.next = 4;
          return webapi.fetchData();

        case 4:
          data = _context.sent;
          this.setState({
            todo: data
          });

        case 7:
        case \\"end\\":
          return _context.stop();
      }
    }
  }, _callee, this);
}));
var _onLoad1 = _asyncToGenerator(
/*#__PURE__*/
_regeneratorRuntime.mark(function _callee() {
  var _this = this;

  var data;
  return _regeneratorRuntime.wrap(function _callee$(_context) {
    while (1) {
      switch (_context.prev = _context.next) {
        case 0:
          app.getUserInfo(function (userInfo) {
            _this.setState({
              userInfo: userInfo
            });
          });
          console.log('onLoading...');
          _context.next = 4;
          return webapi.fetchData();

        case 4:
          data = _context.sent;
          this.setState({
            todo: data
          });

        case 7:
        case \\"end\\":
          return _context.stop();
      }
    }
  }, _callee, this);
}));"
`;

exports[`test setData and spliceData 1`] = `
"
//test mixin
const TabMixin = Mixin({
  data: {
    tabIndex: 0
  },

  onHandleTabIndex(tabIndex) {
    this.setState({
      tabIndex
    });
    this.spliceState({});
  }
});

//test plain javascript object
const pojo = {
  data: {
    test: 0
  },

  sayHello() {
    this.setData({});
    this.$spliceData({});
  }
};

//test mue object
Mue({
  data: {
    list: []
  },

  sayHello() {
    this.setState({});
    this.spliceState({});
  }
});

Action({
  sayHello() {
    this.setState({});
    this.spliceState({});
  }
});"
`;
